### PR TITLE
feat: adding economics data on dashboard

### PR DESCRIPTION
**Pull request by pedroferreira1**
image

### CODE REVIEW

**Comment by pedroferreira1** [comment]
In my opinion the solution is not 100% because the supply values (total and circulating) are not being updated on real time.

There are 2 possible solutions there I see (both have problems):

1. Get this information every time we have a height change (bad because we will have requests every time and if we have many explorer pages opened, we will reach 1M lambda requests easily).
2. Add economics information on websocket dashboard data, so it can be updated just like the rest of the data on this page. (bad because it's called every 5 seconds, so the full node will calculate the mined tokens every time. Also we would need to add a hardcoded `premine_released_tokens` value, just like the one in the lambda function.

What are your thoughts @msbrogli @obiyankenobi ? I think we can merge with no real time update but open an issue to handle it soon.

**Comment by msbrogli** [comment]
As the other statistics are updated in real time, I think it is weird to have static values. Maybe we should update it every minute or something like this. What do you think?

File: src/screens/Dashboard.js [hunk 1]
```diff
@@ -6,10 +6,12 @@
  */
 
 import React from 'react';
+import ReactLoading from 'react-loading';
 import { connect } from "react-redux";
-import LineChartRealTime from '../components/LineChartRealTime';
```
Comment (obiyankenobi) [inline src/screens/Dashboard.js:10]: If we're removing the charts, we should do it properly. Maybe we should have a PR just for that, as there are quite a few changes:
1. remove the chart libs from `package.json`;
2. remove files that won't be used anymore, like `components/LineChartRealTime.js`;
3. refactor the redux state. We currently keep an array of the latest data received. I understand we do it for the charts, so there's no need to continue that;

File: src/screens/Dashboard.js [hunk 2]
```diff
@@ -6,10 +6,12 @@
...
-import AreaChartRealTime from '../components/AreaChartRealTime';
 import colors from '../index.scss';
+import helpers from '../utils/helpers';
+import economicsApi from '../api/economics';
```
Comment (obiyankenobi) [inline src/screens/Dashboard.js:13]: Starting a thread to discuss updating or not the economic metrics in realtime.

@pedroferreira1 actually the data updates every second, which I think we should change to 5 or 10 seconds.

I like the second suggestion. Maybe we should cache the mined tokens info on the server, as it only needs to be calculated when there's a new block. That would require a refactor on hathor core.

One other idea is just calling the economics api when the height of the best chain changes. That would keep this 100% realtime and prevent polling. But there would be a call to calculate the mined tokens everytime there's a new block, for each explorer that's open, so maybe caching it as I mentioned above would be better.

File: src/api/economicsAxiosInstance.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+import { ECONOMICS_API_URL } from '../constants.js';
+const axios = require('axios');
+
+const requestClient = () => {
+  const defaultOptions = {
+    baseURL: ECONOMICS_API_URL,
+    headers: {
+      'Content-Type': 'application/json',
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/api/economicsAxiosInstance.js]: Yes, you are right. Fixed it

File: src/screens/Dashboard.js (hunk 1 above)
Comment (pedroferreira1) [reply src/screens/Dashboard.js:10]: I will create an issue to do that in another PR. I guess it's better not to mix those things.

**Comment by obiyankenobi** [comment]
Can we close this PR? I think we will not do it for the moment and we can open an issue to implement it in the future.

I'd say it's better, before making any changes in the explorer, refactoring the storage and removing the charts.

**Comment by pedroferreira1** [comment]
As discussed, this PR will be closed because it still needs to be discussed how the economics data will be updated in real time.

Two issues were created to track the changes here:
- https://github.com/HathorNetwork/hathor-explorer/issues/26
- https://github.com/HathorNetwork/hathor-explorer/issues/29

**Pull request closed without merging**